docs/src/artifacts.md

```diff
@@ -0,0 +1,172 @@
+os = "macos"
+
+    [[c_simple.download]]
+    sha256 = "6c17d9e1dc95ba86ec7462637824afe7a25b8509cc51453f0eb86eda03ed4dc3"
+    url = "https://github.com/JuliaBinaryWrappers/c_simple_jll.jl/releases/download/c_simple+v1.2.3+0/c_simple.v1.2.3.x86_64-apple-darwin14.tar.gz"
+
+[processed_output]
+git-tree-sha1 = "1c223e66f1a8e0fae1f9fcb9d3f2e3ce48a82200"
+```
+
+This `Artifacts.toml` binds three artifacts; one named `socrates`, one named `c_simple` and one named `processed_output`.
+The single required piece of information for an artifact is its `git-tree-sha1`.
+Because artifacts are addressed only by their content hash, the purpose of an `Artifacts.toml` file is to provide metadata about these artifacts, such as binding a human-readable name to a content hash, providing information about where an artifact may be downloaded from, or even binding a single name to multiple hashes, keyed by platform-specific constraints such as operating system or libgfortran version.
+
+## Artifact types and properties
+
+In the above example, the `socrates` artifact showcases a platform-independent artifact with multiple download locations.
+When downloading and installing the `socrates` artifact, URLs will be attempted in-order until one succeeds.
+The `socrates` artifact is marked as `lazy`, which means that it will not be automatically downloaded when the containing package is installed, but rather will be downloaded on-demand when the package first attempts to use it.
+
+The `c_simple` artifact showcases a platform-dependent artifact, where each entry in the `c_simple` array contains keys that help the calling package choose the appropriate download based on the particulars of the host machine.
+Note that each artifact contains both a `git-tree-sha1` and a `sha256` for each download entry.  This is to ensure that the downloaded tarball is secure before attempting to unpack it, as well as enforcing that all tarballs must expand to the same overall tree hash.
+
+The `processed_output` artifact contains no `download` stanza, and so cannot be installed.
+An artifact such as this would be the result of code that was previously run, generating a new artifact and binding the resultant hash to a name within this project.
+
+## Using Artifacts
+
+Artifacts can be manipulated using convenient APIs exposed from the `Pkg.Artifacts` namespace.
+As a motivating example, let us imagine that we are writing a package that needs to load the [Iris machine learning dataset](https://archive.ics.uci.edu/ml/datasets/iris).
+While we could just download the dataset during a build step into the package directory, and many packages currently do precisely this, that has some significant drawbacks:
+
+* First, it modifies the package directory, making package installation stateful, which we'd like to avoid.  In the future, we would like to reach the point where packages can be installed completely read-only, instead of being able to modify themselves after installation.
```

Comment:
```suggestion
* First, it modifies the package directory, making package installation stateful, which we want to avoid.
  In the future, we would like to reach the point where packages can be installed completely read-only, instead of being able to modify themselves after installation.

```